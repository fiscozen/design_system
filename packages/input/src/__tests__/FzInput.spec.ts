import { describe, it } from "vitest";
import { mount } from "@vue/test-utils";
import { FzInput } from "..";

const NUMBER_OF_INPUTS = 1000;

describe.concurrent("FzInput", () => {
  it("renders label", async ({ expect }) => {
    const wrapper = mount(FzInput, {
      props: {
        label: "Label",
      },
      slots: {},
    });

    expect(wrapper.text()).toContain("Label");
  });

  it("renders leftIcon", async ({ expect }) => {
    const wrapper = mount(FzInput, {
      props: {
        label: "Label",
        leftIcon: "calendar-lines",
      },
      slots: {},
    });

    expect(wrapper.find(".fa-calendar-lines")).toBeTruthy();
  });

  it("renders rightIcon", async ({ expect }) => {
    const wrapper = mount(FzInput, {
      props: {
        label: "Label",
        rightIcon: "credit-card",
      },
      slots: {},
    });

    expect(wrapper.find(".fa-credit-card")).toBeTruthy();
  });

  it("renders helpText", async ({ expect }) => {
    const wrapper = mount(FzInput, {
      props: {
        label: "Label",
      },
      slots: {
        helpText: "This is a helper text",
      },
    });

    await wrapper.vm.$nextTick();
    expect(wrapper.text()).toContain("This is a helper text");
  });

  it("renders errorMessage", async ({ expect }) => {
    const wrapper = mount(FzInput, {
      props: {
        label: "Label",
        error: true,
      },
      slots: {
        errorMessage: "This is an error message",
      },
    });

    await wrapper.vm.$nextTick();
    expect(wrapper.text()).toContain("This is an error message");
  });

  it("renders disabled", async ({ expect }) => {
    const wrapper = mount(FzInput, {
      props: {
        label: "Label",
        disabled: true,
      },
      slots: {},
    });

    expect(wrapper.find("input").attributes("disabled")).toBe("");
  });

  it("renders required", async ({ expect }) => {
    const wrapper = mount(FzInput, {
      props: {
        label: "Label",
        required: true,
      },
      slots: {},
    });

    await wrapper.vm.$nextTick();

    expect(wrapper.find("input").attributes("required")).toBe("");
    expect(wrapper.text()).toContain("*");
  });

  it("renders email type", async ({ expect }) => {
    const wrapper = mount(FzInput, {
      props: {
        label: "Label",
        type: "email",
      },
      slots: {},
    });

    expect(wrapper.find("input").attributes("type")).toBe("email");
  });

  it("renders tel type", async ({ expect }) => {
    const wrapper = mount(FzInput, {
      props: {
        label: "Label",
        type: "tel",
      },
      slots: {},
    });

    expect(wrapper.find("input").attributes("type")).toBe("tel");
  });

  it("renders password type", async ({ expect }) => {
    const wrapper = mount(FzInput, {
      props: {
        label: "Label",
        type: "password",
      },
      slots: {},
    });

    expect(wrapper.find("input").attributes("type")).toBe("password");
  });

  it(`renders ${NUMBER_OF_INPUTS} input with different ids`, async ({
    expect,
  }) => {
    const wrapperList = Array.from({ length: NUMBER_OF_INPUTS }).map((_, i) =>
      mount(FzInput, {
        props: {
          label: `Label ${i}`,
        },
        slots: {},
      }),
    );

    await Promise.all(wrapperList.map((w) => w.vm.$nextTick()));

    const ids = wrapperList.map((w) => w.find("input").attributes("id"));

    expect(new Set(ids).size).toBe(NUMBER_OF_INPUTS);
  });

  it("emits fzinput:right-icon-click event", async ({ expect }) => {
    const wrapper = mount(FzInput, {
      props: {
        label: "Label",
        rightIcon: "eye",
      },
      slots: {},
    });

    await wrapper.find(".fa-eye").trigger("click");

    expect(wrapper.emitted("fzinput:right-icon-click")).toBeTruthy();
  });

  it("emits fzinput:left-icon-click event", async ({ expect }) => {
    const wrapper = mount(FzInput, {
      props: {
        label: "Label",
        leftIcon: "eye",
      },
      slots: {},
    });

    await wrapper.find(".fa-eye").trigger("click");

    expect(wrapper.emitted("fzinput:left-icon-click")).toBeTruthy();
  });
});
