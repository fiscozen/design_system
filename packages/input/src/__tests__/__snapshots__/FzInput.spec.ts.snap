// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FzInput > matches snaphost 1`] = `
"<div class="w-full flex flex-col gap-8"><label class="text-sm text-core-black">Label</label>
  <div class="flex justify-between w-full items-center px-10 border-1 rounded gap-8 text-left has-[:focus]:border-blue-600 h-32 text-base border-grey-300 bg-white text-core-black cursor-pointer">
    <!--v-if--><input type="text" class="peer w-full bg-transparent border-0 focus:outline-none focus:ring-0 cursor-pointer truncate">
    <!--v-if-->
    <!--v-if-->
  </div>
  <!--v-if-->
</div>"
`;
